component/azstorage/azstorage.go

```diff
@@ -331,6 +333,10 @@ func (az *AzStorage) StreamDir(options internal.StreamDirOptions) ([]*internal.O
 	// increment streamdir call count
 	azStatsCollector.UpdateStats(stats_manager.Increment, streamDir, (int64)(1))
 
+	//check for filters provided
+	if len(az.stConfig.blobFilter) > 0 { //only apply filter if user has given
+		new_list = filter.ApplyFilterOnBlobs(new_list)
```

Comment:
Shall we keep the input and output variables same ? just think it might not be a safer way to do conversion.